File: tools/bench/answers/go_haiku/s1/12_celsius.go
package main

import "fmt"

func main() {
	temps := []float64{12.5, 30.0, -5.0}
	max := 0.0
	for i, c := range temps {
		f := c*9.0/5.0 + 32.0
		if i == 0 || f > max {
			max = f
		}
	}
	fmt.Println(max)
}

File: tools/bench/answers/rust_sonnet/s3/12_celsius.rs
fn main() {
    let temps = [12.5f64, 30.0f64, -5.0f64];
    let max_f = temps.iter().map(|&c| c * 9.0 / 5.0 + 32.0).fold(f64::NEG_INFINITY, f64::max);
    println!("{}", max_f);
}
